src/frontend-samples/query-volume-app/VolumeQueryUI.tsx

```diff
@@ -0,0 +1,287 @@
+  spatialElementsColors: SpatialElementsColors;
+  elementsToShow: { elements: PhysicalElement[], position: ElementPosition };
+}
+
+export default class VolumeQueryUI extends React.Component<
+  VolumeQueryUIProps,
+  VolumeQueryUIState
+  > {
+  constructor(props?: any, context?: any) {
+    super(props, context);
+    this.state = {
+      isVolumeBoxOn: false,
+      isClipVolumeOn: false,
+      spatialElements: {} as SpatialElements,
+      spatialElementsColors: { insideColor: ColorDef.green, outsideColor: ColorDef.red, overlapColor: ColorDef.blue },
+      elementsToShow: { elements: [], position: ElementPosition.InsideTheBox },
+    };
+  }
+
+  /* Turining Volume Box on and off  */
+  private _onToggleVolumeBox = (isToggleOn: boolean) => {
+    this.setState({ isVolumeBoxOn: isToggleOn });
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp) {
+      return false;
+    }
+
+    if (isToggleOn) {
+      if (!vp.view.getViewClip()) {
+        this._emptyListOfElements();
+        VolumeQueryApp.clearColoredSelection(vp);
+        VolumeQueryApp.addBoxRange(vp);
+      }
+    } else {
+      VolumeQueryApp.clearClips(vp);
+      this.setState({ isClipVolumeOn: false });
+    }
+    return true;
+  }
+
+  /* Turning Clip Volume feature on and off */
+  private _onToggleClipping = (isToggleOn: boolean) => {
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp) {
+      return false;
+    }
+    this.setState({ isClipVolumeOn: isToggleOn });
+    const range = VolumeQueryApp.getRangeOfBox(vp);
+    VolumeQueryApp.clearClips(vp);
+    VolumeQueryApp.addBoxRange(vp, range, isToggleOn);
+    /* If Volume box is off - turning it on */
+    if (isToggleOn && !this.state.isVolumeBoxOn)
+      this.setState({ isVolumeBoxOn: true });
+  }
+
+  /* Coloring elements that are inside, outside the box or overlaping */
+  private _onClickApplyColoring = async () => {
+    const vp = IModelApp.viewManager.selectedView;
+
+    if (undefined === vp) {
+      return false;
+    }
+    /* Clearing colors so they don't stack when pressing apply button multiple times */
+    VolumeQueryApp.clearColoredSelection(vp);
+
+    /* Getting elements that are going to be colored */
+    const candidates = await VolumeQueryApp.getAllSpacialElements(vp);
+    const spatialElements = await VolumeQueryApp.getSortedSpatialElements(vp, candidates) as SpatialElements;
+    await VolumeQueryApp.colorSpatialElements(vp, spatialElements, this.state.spatialElementsColors);
+    this.setState({ spatialElements });
+    await this._setListOfElements(vp);
+  }
+
+  private _onClickClearColoring = () => {
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp) {
+      return false;
+    }
+
+    VolumeQueryApp.clearColoredSelection(vp);
+    /* Emptying elements to show list*/
+    this._emptyListOfElements();
+  }
+
+  /* Changing colors of elements that are going to be selected */
```

Comment:
The word selected is wrong here.   Selected refers to the user selecting elements, ex. using the element selection tool.  This is changing the colors that will be used for overrides.